client/src/shared/ui/form/FileInput.js

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.
+ *
+ * Camunda licenses this file to you under the MIT; you may not use this file
+ * except in compliance with the MIT License.
+ */
+
+import React from 'react';
+
+import { uniqueBy } from 'min-dash';
+
+import { getIn } from 'formik';
+import classNames from 'classnames';
+
+import CloseIcon from '../../../../resources/icons/Close.svg';
+import FormFeedback from './FormFeedback';
+
+/**
+ * @typedef FileDescriptor
+ * @property {Uint8Array|Blob|null} contents
+ * @property {string} name
+ * @property {string} path
+ */
+
+export default function FileInput(props) {
+  const {
+    label,
+    field,
+    form
+  } = props;
+
+  const {
+    name,
+    value,
+    onBlur
+  } = field;
+
+  const inputRef = React.useRef(null);
+
+  function onChange() {
+    const { files } = inputRef.current;
+    const fileDescriptors = toFileDescriptors(files);
+
+    form.setFieldValue(name, uniqueBy('path', value, fileDescriptors));
+  }
+
+  function removeFile(fileToRemove) {
+    form.setFieldValue(name, field.value.filter(file => file !== fileToRemove));
+  }
+
+  return (
+    <div className="form-group">
+      <input
+        className="form-control"
+        name={ name }
+        id={ name }
+        onBlur={ onBlur }
+        onChange={ onChange }
+        multiple
+        type="file"
+        ref={ inputRef }
+      />
+
+      <label className="btn" htmlFor={ name }>
+        {label}
+      </label>
+
+      <FileList
+        errors={ form.errors }
+        fieldName={ name }
+        files={ value }
+        onRemove={ removeFile }
+      />
+    </div>
+  );
+}
+
+function FileList(props) {
+  const {
+    errors: formErrors,
+    fieldName,
+    files,
+    onRemove
+  } = props;
+
+  const invalid = !!getIn(formErrors, fieldName);
+
+  return (
+    <ul className={ classNames('file-list', { 'is-invalid': invalid }) }>
+      { files.map((file, index) => (
+        <ListItem
+          key={ file.path } name={ file.name } onRemove={ () => onRemove(file) }
+          error={ getIn(formErrors, `${fieldName}[${index}]`) } />
+      ))}
+    </ul>
+  );
+}
+
+function ListItem(props) {
+  const { error, name, onRemove } = props;
+
+  return (
+    <li className={ classNames('file-list-item', { 'is-invalid': !!error }) }>
+      <span className="file-list-item-content">
+        { name }
+        { error ? <span className="error">{'- ' + error}</span> : null }
```

Comment:
We have the error message now placed twice

![image](https://user-images.githubusercontent.com/9433996/111758196-7b4f4380-889c-11eb-8b00-b273d8412061.png)

just to double-check, we agreed on this because the icon might be not visible in some cases?

Reply:
We are generally heading towards displaying the error message instead of hiding them behind hover state. I believe this way it's more readable and would prefer to not use tooltips at all.

@andreasgeier @volkergersabeck I think your opinion is important here as well.

Reply:
To be consistent with the validation of other fields within the modal, we use the icon and the hover hint. But in a scrollable list, the hint could be out of the viewport (based on the length of the list and the scroll position). To ensure that a user would not hover over the icon without any visible reaction we decided to add an inline message as well.

For me, the current (intermediate) solution is okay, although it doubles the warning information.

Reply:
I'm fine with that 👍